base/parse-type.jl

```diff
@@ -0,0 +1,166 @@
+# Parses a string into a Julia type object, e.g. `Int`, `Array{Int, 2}`, etc.
+function Base.parse(::Type{T}, str::AbstractString) where T<:Type
+    ast = try
+        Meta.parse(str)
+    catch
+        throw(ArgumentError("invalid type expression: $str"))
+    end
+    if ast isa Expr && ast.head === :incomplete
+        throw(ArgumentError("invalid type expression: $str"))
+    end
```

Comment:
Shouldn't these checks for a valid `ast` go into `_try_parse_type`?

Reply:
fixed thanks

Reply:
Oh, wait, no: I left this out of that since `_try_parse_type` is called recursively, but this only needs to be checked once. Lemme add a comment to explain that.